pkg/workflows/management/core_components.go

```diff
@@ -0,0 +1,110 @@
+package management
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/types"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type ensureEtcdCAPIComponentsExist struct{}
+
+// Run ensureEtcdCAPIComponentsExist ensures ETCD CAPI providers on the management cluster.
+func (s *ensureEtcdCAPIComponentsExist) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	logger.Info("Ensuring etcd CAPI providers exist on management cluster before upgrade")
+	if err := commandContext.CAPIManager.EnsureEtcdProvidersInstallation(ctx, commandContext.ManagementCluster, commandContext.Provider, commandContext.CurrentClusterSpec); err != nil {
+		commandContext.SetError(err)
+		return &workflows.CollectMgmtClusterDiagnosticsTask{}
+	}
+	return &pauseGitOpsReconcile{}
+}
+
+func (s *ensureEtcdCAPIComponentsExist) Name() string {
+	return "ensure-etcd-capi-components-exist"
+}
+
+func (s *ensureEtcdCAPIComponentsExist) Checkpoint() *task.CompletedTask {
+	return &task.CompletedTask{
+		Checkpoint: nil,
+	}
+}
+
+func (s *ensureEtcdCAPIComponentsExist) Restore(ctx context.Context, commandContext *task.CommandContext, completedTask *task.CompletedTask) (task.Task, error) {
+	return &pauseGitOpsReconcile{}, nil
+}
+
+type upgradeCoreComponents struct {
+	UpgradeChangeDiff *types.ChangeDiff
+}
+
+// Run upgradeCoreComponents upgrades pre cluster upgrade components.
+func (s *upgradeCoreComponents) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
```

Comment:
Where is it upgrading cilium? That's also part of core components upgrade right? I didn't see it in this PR